app/controllers/smart_proxies_controller.rb

```diff
@@ -67,11 +64,25 @@ def destroy
 
   private
 
+  def requested_data(data_name)
+    data = @smart_proxy.public_send(data_name)
+    render :json => {:success => true, :message => data}
+  rescue Foreman::Exception => exception
```

Comment:
I think it should be added to `HTTP_ERRORS` as it is supposed to catch all connection issues and raise one `Foreman::Exception`. Easier to read and parse ("One error to rule them all")
![image](https://cloud.githubusercontent.com/assets/433583/11976877/4560e5fe-a985-11e5-9495-79db88f4a6e2.png)